numpy/random/tests/test_random.py

```diff
@@ -1421,6 +1421,17 @@ def test_hypergeometric(self):
         assert_raises(ValueError, hypergeom, ngood, nbad, bad_nsample_one * 3)
         assert_raises(ValueError, hypergeom, ngood, nbad, bad_nsample_two * 3)
 
+    def test_hypergeometric_propogates_exceptions(self):
+        # Tests that hypergeometric correctly propagates exceptions
+        # when called with a type which throws when converted to an int
+
+        class ThrowingInteger(np.ndarray):
+            def __int__(self):
+                raise TypeError
```

Comment:
Thanks for doing this, just a minor nitpick...

Someone seeing this test and the one you wrote for `PyFloat_AsDouble` may think there's something fundamentally different between both cases since the exceptions are different. So it would probably be good to make both more equal by having both raise the same error.

I also think it would be good to group them both in a `test_scalar_exception_propagation` method, rather than have them grouped with the random method used for testing, but this may be a matter of personal taste.